app/dftb+/dftbplus.F90

```diff
@@ -22,17 +22,19 @@ program dftbplus
 
   type(TEnvironment) :: env
   type(TInputData), allocatable :: input
-  type(TDftbPlusMain) :: main
+  type(TDftbPlusMain), allocatable, target :: main
 
   call initGlobalEnv()
   call printDftbHeader(releaseName, releaseYear)
   allocate(input)
   call parseHsdInput(input)
   call TEnvironment_init(env)
+  allocate(main)
   call main%initProgramVariables(input, env)
   deallocate(input)
   call runDftbPlus(main, env)
   call main%destructProgramVariables()
+  deallocate(main)
```

Comment:
I suspect this will also help with finalizing other things, so definitely should get merged.

Reply:
Actually, we still have memory leaks (despite this), part of that will be fixed in a followup PR.